kratos/geometries/geometry_data.h

```diff
@@ -438,6 +504,17 @@ class GeometryData
     ///@name Shape Function
     ///@{
 
+
+    const ShapeFunctionsDerivativesIntegrationPointsType& ShapeFunctionsDerivativesIntegrationPoints(IndexType DerivativeOrder) const
+    {
+        return mShapeFunctionsDerivativesContainer[mDefaultMethod][DerivativeOrder];
```

Comment:
```suggestion
        return mShapeFunctionsDerivativesContainer[DerivativeOrder][mDefaultMethod];
```
defining the container this way would simplify the implementation. Then you could replace `mShapeFunctionsLocalGradients` with mShapeFunctionsDerivativesContainer[1];`

Reply:
I think like this it could be the same. One never returns the container. Thus, would just be mShapeFunctionsLocalGradients = mShapeFunctionsLocalGradients[mDefaultMethod][1]. Anyways, I can live with both.

Reply:
I would not do `mShapeFunctionsLocalGradients = mShapeFunctionsLocalGradients[mDefaultMethod][1]` as it is wastes memory and you need to synchronize data.